applications/matter_bridge/src/ble_providers/ble_bridged_device_factory.cpp

```diff
@@ -0,0 +1,428 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "ble_bridged_device_factory.h"
+#include "bridge_manager.h"
+#include "bridge_storage_manager.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+namespace
+{
+/* Maps single Matter device type to corresponding BLE service. */
+CHIP_ERROR MatterDeviceTypeToBleService(MatterBridgedDevice::DeviceType deviceType,
+					BleBridgedDeviceFactory::ServiceUuid &serviceUUid)
+{
+	switch (deviceType) {
+	case MatterBridgedDevice::DeviceType::OnOffLight:
+		serviceUUid = BleBridgedDeviceFactory::ServiceUuid::LedButtonService;
+		break;
+	case MatterBridgedDevice::DeviceType::TemperatureSensor:
+	case MatterBridgedDevice::DeviceType::HumiditySensor:
+		serviceUUid = BleBridgedDeviceFactory::ServiceUuid::EnvironmentalSensorService;
+		break;
+	default:
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+	return CHIP_NO_ERROR;
+}
+
+/* Maps single BLE service to one or multiple Matter device types. */
+CHIP_ERROR BleServiceToMatterDeviceType(BleBridgedDeviceFactory::ServiceUuid serviceUUid,
+					MatterBridgedDevice::DeviceType deviceType[], uint8_t maxCount, uint8_t &count)
+{
+	switch (serviceUUid) {
+	case BleBridgedDeviceFactory::ServiceUuid::LedButtonService: {
+		if (maxCount == 0) {
+			return CHIP_ERROR_BUFFER_TOO_SMALL;
+		}
+		deviceType[0] = MatterBridgedDevice::DeviceType::OnOffLight;
+		count = 1;
+	} break;
+	case BleBridgedDeviceFactory::ServiceUuid::EnvironmentalSensorService: {
+		if (maxCount < 2) {
+			return CHIP_ERROR_BUFFER_TOO_SMALL;
+		}
+		deviceType[0] = MatterBridgedDevice::DeviceType::TemperatureSensor;
```

Comment:
NIT: If we use this method to get the Matter Device Type, how we will know that the Temperature Sensor is at index 0 and the Humidity Sensor is at index 1 without looking at the code of this method?

Reply:
And why we would like to know it? It's an internal method to help mapping one type to another, so we would know what device types should be created and assigned to endpoints. I don't think the order of indexes really matter.